### ISSUE TITLE
[Technical Initiative Funding Request]: RSTUF Security Audit for 1.0.0

### ISSUE THREAD

**Issue by kairoaraujo** [ossf/tac#379]
### Technical Initiative

Repository Service for TUF / Security Software Repositories Working Group

### Lifecycle Phase

incubation

### Funding amount

unknown -- help needed

### Problem Statement

RSTUF is about to release the 1.0.0, we are moving close to 1.0.0-rc and before the final release we aim to have a security audit. This project is moving to staging phase in the PyPI and RubyGems repository.

### Who does this affect?

Public Repository (and or Private repositories)

### Have there been previous attempts to resolve the problem?

No

### Why should it be tackled now and by this TI?

RSTUF is a project under wg-securing-software-repos (OpenSSF)

### Give an idea of what is required to make the funding initiative happen

We require a security audit in the RSTUF projects, to help us to identify possible improvements on this aspect.

### What is going to be needed to deliver this funding initiative?

A company/contractor that can run the audit and generate a report for RSTUF maintainers.

### Give a summary of the requirements that contextualize the costs of the funding initiative

- Meeting with RSTUF Maintainers
- Security Audit
- Security Report

### Who is responsible for doing the work of this funding initiative?

Kairo de Araujo (@kairoaraujo)

### Who is accountable for doing the work of this funding initiative?

Kairo de Araujo (@kairoaraujo)

### If the responsible or accountable parties are no longer available, what is the backup contact or plan?

Martin Vrachev (@MVrachev)

### What license is this funding initiative being used under?

MIT

### Code of Conduct

- [X] I agree to follow the OpenSSF's Code of Conduct

### List the major milestones by date and identify the overall timeline within which the technical initiative plans to accomplish their goals. Any payments for services, sponsorships, etc., will require LF Legal and Financial review.

By the end of Q4 2024 we want to have the RSTUF 1.0.0 out.

**Comment by steiza** [comment]
Hey @kairoaraujo thanks for submitting this request! [This week is the TAC review period for funding request](https://github.com/ossf/tac/blob/main/process/TI%20Funding%20Request%20Process.md#q3-2024), but we need a specific amount to review this request. Have you had a chance to reach out to folks who might perform the audit to get an idea of the amount needed?

**Comment by kairoaraujo** [comment]
> Hey @kairoaraujo thanks for submitting this request! [This week is the TAC review period for funding request](https://github.com/ossf/tac/blob/main/process/TI%20Funding%20Request%20Process.md#q3-2024), but we need a specific amount to review this request. Have you had a chance to reach out to folks who might perform the audit to get an idea of the amount needed?

Hi @steiza, I requested a quotation for the audit.
I will update the issue as soon I have it.

**Comment by Amir-Montazery** [comment]
Hello. The Open Source Technology Improvement Fund, Inc specializes in exactly this: facilitating and managing security audits for open source projects and communities. We did the php TUF (https://ostif.org/php-tuf-audit-complete/), go TUF (https://ostif.org/go-tuf-on-bugs-ostifs-audit-of-go-tuf/), and python TUF (https://ostif.org/our-audit-of-python-tuf-is-complete-multiple-issues-found-and-fixed/) security audits. So in addition to being an ideal candidate to handle any security audit request, OSTIF has a track record for doing these specific kinds of engagements.  I hope you consider engaging us for this request, as we can deliver the intended results.

**Comment by riaankleinhans** [comment]
Gitvote was added as a tool to test for stream lining the TI Funding process.
The members of the GH group "TAC" can vote by commenting with an +1. -1 or eye on the Gitvote block in this issue.
Until the TAC is satisfied with the process the GitVote outcome would not be binding.

Community members can show their support by also voting, however only the "TAC" GH Group's votes will count.

The current passing threshold is 70% and the committee is the [TAG GH group](https://github.com/orgs/ossf/teams/tac).
The vote say open fo 6 week and an announcement is sent on the GH/TAC/Discussion

All these parameters can by fine tuned or changed [here](https://github.com/ossf/tac/blob/main/.gitvote.yml)
Please reach out if you have any questions.

**Comment by marcelamelara** [comment]
>  I requested a quotation for the audit. I will update the issue as soon I have it.

@kairoaraujo pinging to see if you have an update on this?

**Comment by kairoaraujo** [comment]
> @kairoaraujo pinging to see if you have an update on this?

I had one meeting with one provider yesterday (23rd Sept) and I have another one today (24th Sept).
I will have an update soon on this issue.

**Comment by SecurityCRob** [comment]
A quorum of the TAC met on 17Sept to discuss Q3 TI Funding Requests.

The group reached consensus that pending a quote for the review that this is tentatively approved.

**Comment by sevansdell** [comment]
I see the vote in progress is cancelled. Is the request still open?

**Comment by steiza** [comment]
> Is the request still open?

Yes, a quorum of the TAC approved this on September 17th: https://github.com/ossf/tac/issues/379#issuecomment-2371347503

This was one of the two issues I was asking @riaankleinhans about at the TAC meeting yesterday. On the [funding project board](https://github.com/orgs/ossf/projects/25/views/1) it's sitting in `Funding Approved` but hasn't yet moved to `Funding in Execution`.

**Comment by sevansdell** [comment]
@kairoaraujo have you received the other quotes?

**Comment by Amir-Montazery** [comment]
Hello everyone, chiming in here. OpenSSF has asked OSTIF to take a look at the proposal and help with moving forward with the security audit. We will take a look and explore next steps.

**Comment by riaankleinhans** [comment]
@kairoaraujo I will send out an email for formalized the process between OpenSSF & OSTIF.

**Comment by riaankleinhans** [comment]
@kairoaraujo as the process between RSTUF & OSTIF is in progress and SOW have been received which is well with in the approved budget, can this issue be closed?

**Comment by Naomi-Wash** [comment]
Hello everyone - we've created the signed agreement and the funds are being released.
